core/src/main/java/org/keycloak/representations/idm/GroupRepresentation.java

```diff
@@ -17,23 +17,35 @@
 
 package org.keycloak.representations.idm;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
 public class GroupRepresentation {
+    // For an individual group these are the sufficient minimum fields
+    // to identify a group and operate on it in a basic way
     protected String id;
     protected String name;
     protected String path;
+    protected String parentId;
+    protected Long subGroupCount;
+    // For navigating a hierarchy of groups, we can also include a minimum representation of subGroups
+    // These aren't populated by default and are only included as-needed
+    protected List<GroupRepresentation> subGroups = new ArrayList<>();
```

Comment:
I initially added this because the code that uses it in a lot places is not null safe. This wasn't an issue before because a GroupRepresentation always had a subGroups list associated with it but now that isn't the case. 

Will look at the singleton option and worst case patch the existing code to be null safe!